google_shopping.rb

```diff
@@ -12,4 +12,52 @@
 puts google_data.inspect
 
 #outputs the first item's title
-puts google_data["items"][0]["product"]["title"]
\ No newline at end of file
+puts google_data["items"][0]["product"]["title"]
+
+#outputs number of items that have a kind of "shopping#product"
+ i = 0
+ for i in (0...items.length)
+   if items[i]['kind'] === 'shopping#product'
+   shopping_products += 1
+   end
+ end
+ 
+ #outputs products availbility that are 'backorder'
+ items.each do |item|
+   if item["product"]["inventories"][0]["availability"] ==='backorder'
+     puts item["product"]["title"]
+   end
+ end
+ 
+ # outputs items with more than one image link
+ items.each do |item|
+   if item["product"]["images"][0]["link"].length > 1
+     puts item["product"]["title"]
+   end
+ end
+
+ 
+ #outputs all items that are Canon products regardless of case sensitivity
+items.each do |item|
+   if item["product"]["brand"].downcase === 'canon'
+     puts item["product"]["title"]
+   end
+ end
+ 
+ #outputs all items that are Canon products with an author name of eBay
+ items.each do |item|
+  if item["product"]["brand"].downcase === 'canon'
```

Comment:
We can also use the ```&&``` operator to avoid using a nested if statement here.
```
 if item["product"]["brand"].downcase === 'canon' && item["product"]["author"]["name"].downcase.include?("ebay")
```